File: IfeedsApi/Api/Models/RespostaFeedbackModel.cs
using AutoMapper;
using IfeedsApi.Domain.Models;

namespace IfeedsApi.Api.Models
{
    [AutoMap(typeof(RespostaFeedback), ReverseMap = true)]
    public class RespostaFeedbackModel
    {
        public int Id { get; set; }
        public string Titulo { get; set; }
        public string Descricao { get; set; }
        public decimal Nota { get; set; }
        public string Resposta { get; set; }
        public string Codigo { get; set; }
        public int FeedbackId { get; set; }
        public bool Status { get; set; }
        public string Usuario { get; set; }
    }

    [AutoMap(typeof(RespostaFeedback), ReverseMap = true)]
    public class RespostaFeedbackModelRequest
    {
        public string Resposta { get; set; }
        public int FeedbackId { get; set; }
        public int UsuarioId { get; set; }
    }

    [AutoMap(typeof(RespostaFeedback), ReverseMap = true)]
    public class RespostaFeedbackModelUpdateRequest
    {
        public string Resposta { get; set; }
    }
}

File: ifeeds_app/lib/models/resposta_feedback_model.dart
import 'dart:convert';

class RespostaFeedbackModel {
  int? id;
  String? titulo;
  String? descricao;
  String? resposta;
  String? codigo;
  int? feedbackId;
  double? nota;
  bool? status;
  String? usuario;

  RespostaFeedbackModel({
    this.id,
    this.titulo,
    this.descricao,
    this.resposta,
    this.codigo,
    this.feedbackId,
    this.nota,
    this.status,
    this.usuario,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'titulo': titulo,
      'descricao': descricao,
      'resposta': resposta,
      'codigo': codigo,
      'feedbackId': feedbackId,
      'nota': nota,
      'status': status,
      'usuario': usuario,
    };
  }

  factory RespostaFeedbackModel.fromMap(Map<String, dynamic> map) {
    return RespostaFeedbackModel(
      id: map['id'],
      titulo: map['titulo'],
      descricao: map['descricao'],
      resposta: map['resposta'],
      codigo: map['codigo'],
      feedbackId: map['feedbackId'],
      nota: map['nota'],
      status: map['status'],
      usuario: map['usuario'],
    );
  }

  String toJson() => json.encode(toMap());

  factory RespostaFeedbackModel.fromJson(String source) =>
      RespostaFeedbackModel.fromMap(json.decode(source));
}
